install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -99,22 +89,18 @@ spec:
           exec:
             command:
             - "true"
-{{- else }}
-      - args:
-        - --config-dir=/tmp/cilium/config-map
-{{- with .Values.extraArgs }}
-        {{- toYaml . | trim | nindent 8 }}
-{{- end }}
+        {{- else }}
         command:
         - cilium-agent
-{{- if semverCompare ">=1.20-0" $k8sVersion }}
+        args:
+        - --config-dir=/tmp/cilium/config-map
+        {{- with .Values.extraArgs }}
+        {{- toYaml . | trim | nindent 8 }}
+        {{- end }}
+        {{- if semverCompare ">=1.20-0" .Capabilities.KubeVersion.Version }}
```

Comment:
As mentioned above, we do want to use `$k8sVersion` here, not `.Capabilities.KubeVersion.Version`